### ISSUE TITLE
ld: warning: object file (...) was built for newer OSX version (10.12) than being linked (10.10)

### ISSUE THREAD

**Issue by kdawgwilk** [harlanhaskins/LLVMSwift#6]
On a fresh project with LLVM installed through homebrew and pkg-config file added I get the following warnings when running `swift build`:

```
$ swift build
Cloning https://github.com/harlanhaskins/LLVMSwift.git
HEAD is now at 0a03385 Added handy function for creating a global string constant
Resolved version: 0.0.3
Cloning https://github.com/harlanhaskins/cllvm.git
HEAD is now at 092074b Initial commit
Resolved version: 0.0.1
Compile Swift Module 'LLVMSwift' (13 sources)
Compile Swift Module 'MyLanguage' (1 sources)
Linking ./.build/debug/MyLanguage
ld: warning: object file (/usr/local/opt/llvm/lib/libLLVMScalarOpts.a(Scalar.cpp.o)) was built for newer OSX version (10.12) than being linked (10.10)
...
ld: warning: object file (/usr/local/opt/llvm/lib/libLLVMSupport.a(regstrlcpy.c.o)) was built for newer OSX version (10.12) than being linked (10.10)
```

Is this normal?  Or can I do something to resolve the warnings?

**Comment by harlanhaskins** [comment]
Unfortunately, this is an artifact of statically linking the LLVM binaries. I talked with @aciidb0mb3r about what to do for this, and basically we can't do much until LLVM fixes their dylib file and we can use that instead.

**Comment by CodaFi** [comment]
If you want an easy way to silence this (assuming you're running on 10.12+) just set your deployment targets to 10.12.

**Comment by CodaFi** [comment]
Just to update you on this.  The Swift 5.0 version of package manifests allows us to impose deployment target constraints.  We are beginning to do so in #207, and will continue to update our deployment target to follow the LLVM binaries built through homebrew.

**Comment by CodaFi** [comment]
tl;dr No more warnings.

**Comment by matthewseaman** [comment]
Unfortunately, it seems SPM's deployment target customization only changes the build settings for each target, not for the product. The result is that packages using something like LLVMSwift as a dependency still need to manually change the deployment target for the product in Xcode, even if the deployment target is overridden in the client package manifest. (the warnings still show up unless the product's target is customized)

**Comment by CodaFi** [comment]
> Unfortunately, it seems SPM's deployment target customization only changes the build settings for each target, not for the product. 

That sounds like a bug.  I just tried it with silt, and the generated Xcode project had all the right settings and I was able to build the thing just fine.

If we are used as a dependency in a situation where deployment targets aren't set forward far enough, the result should be a linker error.  I suspect there's some deeper shenanigans going on.  You could try to regenerate your project file with 

```
$ swift package update
$ swift package generate-xcodeproj
```
